### ISSUE TITLE
Method for dropping an element of NamedTuple

### ISSUE THREAD

**Issue by Datseris** [JuliaLang/julia#27574]
I am trying to do:
```
d = (solver = 2, b = 3)
f = same tuple as d but without :solver 
```

The [JuliaData/NamedTuples](https://github.com/JuliaData/NamedTuples.jl) has a `delete` method, so one can do something like `f = delete(d, :solver)`. It would be nice to have it for `Core.NamedTuple` as well, if possible.

**Comment by Datseris** [comment]
Current workaround:

```julia
julia> d = (solver = 2, b = 3, c = 4)
(solver = 2, b = 3, c = 4)

julia> inter = [p for p in pairs(d) if p[1] != :solver]
2-element Array{Pair{Symbol,Int64},1}:
 :b => 3
 :c => 4

julia> f = (; inter...)
(b = 3, c = 4)
```
(answered on Slack, I think by araslan)

**Comment by JeffreySarnoff** [comment]
```julia

julia> function dropnames(namedtuple::T, names::Tuple) where {T<:NamedTuple} 
           keepnames = (setdiff(Base._nt_names(d), names)...,)
           return NamedTuple{keepnames}(namedtuple)
       end
dropnames (generic function with 2 methods)

julia> dropnames(d, (:solver,))
(b = 3,)

julia> dropnames(d, (:b,))
(solver = 2,)

julia> dropnames(d, (:b,:solver))
NamedTuple()

julia> dropnames(d, (:oops,))
(solver = 2, b = 3)

```

**Comment by Datseris** [comment]
Very nice! Can it also become part of Base or I am being too greedy?

**Comment by JeffreySarnoff** [comment]
Base has my permission.

**Comment by chethega** [comment]
Is the inference failure OK (everything becomes `Any`)? 

I would imagine that most use-cases could get away with `@generated` and val-type symbols (the symbol to drop is a source-code literal).  On the other hand, I'd imagine that most uses are not performance critical.

**Comment by JeffreySarnoff** [comment]
`dropnames` implementation improved ^^

**Comment by JeffreySarnoff** [comment]
I don't know what to do about this:
```julia
julia> code_warntype(dropnames, (NamedTuple, Tuple{Vararg{Symbol}}))
Body::Any
2 1 ─ %1 = Base._nt_names::typeof(Base._nt_names)                                           │
  │   %2 = %1(%%namedtuple)::Any                                                            │
  │        Core.typeassert(%2, Tuple{Vararg{Symbol,N} where N})                             │
  │   %4 = π (%2, Tuple{Vararg{Symbol,N} where N})                                          │
3 │   %5 = Core.apply_type(Main.NamedTuple, %4)::Type{NamedTuple{_1,T} where T<:Tuple} where _1
  │   %6 = %5(%%namedtuple)::Any                                                            │
  └──      return %6    
```

**Comment by JeffBezanson** [comment]
That happens because `NamedTuple` is too abstract.

**Comment by brenhinkeller** [comment]
Looks like the PR that implemented this got approved after some debate about naming, but then never merged

**Comment by JeffreySarnoff** [comment]
+1 reopening and utilizing [that PR](https://github.com/JuliaLang/julia/pull/27725) 
     _if more edits are needed, I will do them_

**Comment by brenhinkeller** [comment]
Adding triage label, see https://github.com/JuliaLang/julia/pull/27725#issuecomment-1690735876

**Comment by LilithHafner** [comment]
Triage discussed this (summary: yes) https://github.com/JuliaLang/julia/pull/27725#issuecomment-1803095344

**Comment by JeffBezanson** [comment]
closed by #55270
